pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompaction/UpsertCompactionTaskExecutor.java

```diff
@@ -108,7 +110,11 @@ protected SegmentConversionResult convert(PinotTaskConfig pinotTaskConfig, File
             .setSegmentName(segmentName).build();
 
     long endMillis = System.currentTimeMillis();
-    LOGGER.info("Finished task: {} with configs: {}. Total time: {}ms", taskType, configs, (endMillis - startMillis));
+    long validDocIdsCount = validDocIds.stream().count();
```

Comment:
Use `validDocIds.getCardinality()` instead. It is much cheaper than looping it over.

Reply:
Wouldn't this always be the same as `totalDocsAfterCompaction`?

Reply:
yeah it is the same. i originally added it because i noticed we had a case where we were compacting the same segment repeatedly, but i think it had to do with frequent new upserts. i'll remove the extra count.